BedrockServer.cpp

```diff
@@ -1581,7 +1571,7 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
 
         // If `Connection: close` was set, shut down the socket, in case the caller ignores us.
         if (SIEquals(command->request["Connection"], "close") || _shutdownState.load() != RUNNING) {
-            shutdownSocket(command->socket, SHUT_RDWR);
+            command->socket->shutdown();
```

Comment:
I changed this API to make it easier to manage socket state (i.e., everywhere that changes the state of a Socket is actually inside the Socket class).